subprojects/core/src/integTest/groovy/org/gradle/api/tasks/IncrementalTasksIntegrationTest.groovy

```diff
@@ -108,4 +112,70 @@ class FailingIncrementalTasksIntegrationTest extends AbstractIntegrationSpec {
         "remove"      | false       | "with removed outputs is fully rebuilt"
         "none"        | true        | "with unmodified outputs is executed as incremental"
     }
+
+    @Issue("https://github.com/gradle/gradle/issues/9320")
+    def "incremental task with NAME_ONLY input (matching file names and content) detects changed input"() {
+        def taskName = "incrementalTask"
+        def inputs = ('a'..'z').collect { file("${it}/input.txt").createFile() }
+        def modifiedInput = inputs.toList().sort { random() }.first()
+
+        @Language("Groovy") script = """
```

Comment:
I guess the `@Language` annotation injects the `Groovy` language when editing this file in IDEA. I am surprised that the annotation is no the classpath...

We do not use this annotation anywhere else, right? I don't think it is worth introducing it in this place.

Reply:
It helps in interpreting the script, otherwise it's just text:
![image](https://user-images.githubusercontent.com/1841944/57223782-85009280-7007-11e9-96cd-a2adf285874f.png)